### ISSUE TITLE
default coref model performance issue?

### ISSUE THREAD

**Issue by pitrack** [allenai/allennlp#2380]
**Describe the bug**
On both 1) py3.6.8+allennlp0.8.0+pytorch1.0.0 and 2) py3.7.1+v0.8.1+pytorch0.4.1, coref model results in low dev/test `coref_f1` scores with the out-of-the-box `coref.jsonnet` config file, e.g.
(from config 1; cpu/gpu info removed)
```
 Metrics: {
  "training_duration": "05:18:43",
  "training_start_epoch": 0,
  "training_epochs": 50,
  "epoch": 50,
  "training_coref_precision": 0.8096445949078802,
  "training_coref_recall": 0.6529397291206623,
  "training_coref_f1": 0.7224902434739162,
  "training_mention_recall": 0.9410573548129958,
  "training_loss": 30.991527096543813,,
  "validation_coref_precision": 0.6751397335767036,
  "validation_coref_recall": 0.5655598814636882,
  "validation_coref_f1": 0.6151873021850315,
  "validation_mention_recall": 0.9133907595927956,
  "validation_loss": 56.70462350607999,
  "best_epoch": 41,
  "best_validation_coref_precision": 0.6789837388296601,
  "best_validation_coref_recall": 0.565021783370241,
  "best_validation_coref_f1": 0.6164966053605297,
  "best_validation_mention_recall": 0.9135473766640564,
  "best_validation_loss": 56.01940999700859
}
```

or (from config 2)
```
  "test_coref_precision": 0.6737676428692932,
  "test_coref_recall": 0.5689162375270107,
  "test_coref_f1": 0.616446981922579,
  "test_mention_recall": 0.9195506982392229,
  "test_loss": 57.740909122709375
```

**To Reproduce**
Fresh install allennlp+pytorch (in my case, these are in a conda environment; allennlp was pip installed) and run `allennlp train coref.jsonnet` with the conll files as input.

**Expected behavior**
67.2

**System (please complete the following information):**
Both setups provided above; Linux machines with CUDA (9.1).

**Additional context**
See #1545 for similar issue (which gets almost identical numbers!) but the fix should have been incorporated by this version.

It's also possible but unlikely that the cause is due to installation problems, ~~e.g. for me, the whole test suite fails fails on the third test of `find_learning_rate_test` with this error `bin/allennlp: line 2:  8679 Terminated              python -m allennlp.run "$@"`.~~ had gpu config set incorrectly. actual test results only failed on all 3 `TestCustomHighwayLSTM` tests.

**Comment by nelson-liu** [comment]
hmm, i remember from investigating #1545 that the loss should be much lower...will look into it.

**Comment by nelson-liu** [comment]
i must have misremembered, sorry. i think this is actually intentional, this is the output from the pretrained coref model we have (https://allennlp.org/models):

```
2019-01-17 11:11:47,585 - INFO - allennlp.commands.evaluate - Finished evaluating.
2019-01-17 11:11:47,585 - INFO - allennlp.commands.evaluate - Metrics:
2019-01-17 11:11:47,585 - INFO - allennlp.commands.evaluate - coref_precision: 0.6748966176089323
2019-01-17 11:11:47,585 - INFO - allennlp.commands.evaluate - coref_recall: 0.5856540941149043
2019-01-17 11:11:47,585 - INFO - allennlp.commands.evaluate - coref_f1: 0.6269946422170675
2019-01-17 11:11:47,585 - INFO - allennlp.commands.evaluate - mention_recall: 0.9190953248330298
2019-01-17 11:11:47,585 - INFO - allennlp.commands.evaluate - loss: 60.66681790925113
```

it's known that we're off from the 67.2 reported in the elmo paper, see https://github.com/allenai/allennlp/blob/master/tutorials/how_to/elmo.md#reproducing-the-results-from-deep-contextualized-word-representations for more details.

**Comment by pitrack** [comment]
Thanks for checking it out! I assumed the other thread resolved to something much higher - I didn't expect the difference between this implementation and the original Lee model to be this large.

**Comment by nelson-liu** [comment]
Yeah, unfortunately the implementation here is missing quite a few features vs. the original code / isn't as memory-efficient, so we need to use less antecedents. closing this now, feel free to reopen if you have further questions.
